src/YGImage.cc

```diff
@@ -10,6 +10,7 @@
 #include "YImage.h"
 #include "ygtkimage.h"
 #include <string.h>
+#include <boost/filesystem.hpp>
 
 static inline bool endsWith (const std::string &str1, const char *str2)
```

Comment:
JFYI: There is also `boost::algorithm::ends_with()` which might take some of the maintenance burden off your shoulders in the future.

https://techoverflow.net/2022/01/25/how-to-use-boost-string-algorithm-ends_with-a-minimal-example/

But it's your choice whether or not you want to use that.